app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala

```diff
@@ -340,51 +326,52 @@ class MovementsRepositoryImpl @Inject() (
       movementMRNFilter(movementReferenceNumber),
       movementLRNFilter(localReferenceNumber)
     )
-    val messageSummaryQuery = Json.obj("$ifNull" -> Json.arr("$movementSummary", "[]"))
-
     val (from, itemCount) = indices(page, count)
-
-    val aggregates = Seq(
-      Aggregates.filter(selector),
+    val filterAggregates  = Seq(Aggregates.filter(selector))
+    val aggregates = filterAggregates ++ Seq(
       Aggregates.sort(descending("updated")),
-      Aggregates.project(Projections.exclude("messages")),
-      Aggregates.facet(
-        Facet("totalCount", Aggregates.count()),
-        Facet("movementSummary", Aggregates.skip(from), Aggregates.limit(itemCount))
-      ),
-      Aggregates.project(Document("totalCount" -> Codecs.toBson(totalCountQuery()), "movementSummary" -> Codecs.toBson(messageSummaryQuery)))
+      Aggregates.skip(from),
+      Aggregates.limit(itemCount)
     )
+    for {
+      perPageMovements <- filterPerPageMovements(aggregates)
+      totalCount       <- totalCountMovements(filterAggregates)
+    } yield PaginationMovementSummary(TotalCount(totalCount), perPageMovements)
+
+  }
 
-    mongoRetry(Try(collection.aggregate[PaginationMovementSummary](aggregates)) match {
+  private def filterPerPageMovements(aggregates: Seq[Bson]): EitherT[Future, MongoError, Vector[MovementWithoutMessages]] =
+    mongoRetry(Try(collection.aggregate[MovementWithoutMessages](aggregates)) match {
       case Success(obs) =>
         obs
           .toFuture()
           .map {
-            response => Right(response.head)
+            response => Right(response.toVector)
           }
 
       case Failure(NonFatal(ex)) =>
         Future.successful(Left(UnexpectedError(Some(ex))))
     })
 
-  }
+  private def totalCountMovements(totalDocument: Seq[Bson]): EitherT[Future, MongoError, Long] =
+    mongoRetry(Try(collection.aggregate[MovementWithoutMessages](totalDocument)) match {
+      case Success(obs) =>
+        obs.toFuture().map {
+          response => Right(response.size.toLong)
```

Comment:
Rather than pulling all the documents into memory and measuring the size of the list, use the `$count` aggregation instead. 

You could put this in this method, so that you pass the filter you want to use in, and this method automatically puts the `count` into the pipeline if you so wished.